src/UserSpaceInstrumentation/RegisterState.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "UserSpaceInstrumentation/RegisterState.h"
+
+#include <cpuid.h>
+#include <elf.h>
+#include <errno.h>
+#include <sys/ptrace.h>
+#include <sys/uio.h>
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "absl/strings/str_format.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace {
+
+// Return the size of the XSave area on this cpu.
+[[nodiscard]] ErrorMessageOr<size_t> GetXSaveAreaSize() {
+  uint32_t eax = 0;
+  uint32_t ebx = 0;
+  uint32_t ecx = 0;
+  uint32_t edx = 0;
+  if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx) || !(ecx & bit_XSAVE)) {
```

Comment:
Could you add a comment somewhere on what `__get_cpuid` and `__get_cpuid_count` do and how they are used?

Reply:
I added something above these two functions. Not sure if this is a good place. Let me know if you have a better idea...